homeassistant/components/sms/gateway.py

```diff
@@ -15,6 +19,120 @@ class Gateway:
     def __init__(self, worker, hass):
         """Initialize the sms gateway."""
         self._worker = worker
+        self._hass = hass
+
+    async def init_async(self):
+        """Initialize the sms gateway asynchronously."""
+        try:
+            await self._worker.set_incoming_sms_async()
+        except gammu.ERR_NOTSUPPORTED:
+            _LOGGER.warning("Your phone does not support incoming SMS notifications!")
+        else:
+            await self._worker.set_incoming_callback_async(self.sms_callback)
+
+    def sms_callback(self, state_machine, callback_type, callback_data):
+        """Receive notification about incoming event.
+
+        @param state_machine: state machine which invoked action
+        @type state_machine: gammu.StateMachine
+        @param callback_type: type of action, one of Call, SMS, CB, USSD
+        @type callback_type: string
+        @param data: event data
+        @type data: hash
+        """
+        _LOGGER.debug(
+            "Received incoming event type:%s,data:%s", callback_type, callback_data
```

Comment:
The log messages are missing a single space after comma and colon. Please follow standard punctuating rules inside the log message. We do not end logging messages with period though.